plugins/inputs/ctrlx_datalayer_sse/README.md

```diff
@@ -0,0 +1,139 @@
+
+## Configuration
+
+Intervals e.g. sampling or publishing interval are durations of time and can be specified for supporting settings by combining an integer value and time unit as a string value. Valid time units are
+`ns`, `us` (or `µs`), `ms`, `s`, `m`, `h`.
+
+```toml
+# A ctrlX Data Layer server sent event input plugin
+[[inputs.ctrlx_datalayer_sse]]
+  ## Hostname or IP address of ctrlX CORE ('localhost' for same device)
+  hostname = "https://localhost"
+
+  ## Username for authentication
+  username = "boschrexroth"
+
+  ## Password for authentication
+  password = "boschrexroth"
+
+  ## Use TLS but skip chain & host verification
+  insecure_skip_verify = true
+
+  ## [optional]  Timeout for HTTP request. (default: "5s").
+  # timeout = "5s"
+
+  ## Create a ctrlX Data Layer SSE subscription
+  [[inputs.ctrlx_datalayer_sse.subscription]]
+    ## Define an array of address to nodes in the ctrlX Data Layer
+    nodes = [
+      "framework/metrics/system/cpu-utilisation-percent",
+      "framework/metrics/system/memavailable-mb",
+      "framework/metrics/system/memused-mb",
+      "framework/metrics/system/memused-percent",
+      "scheduler/tasks/ctrlXAutomation/info/counter"
+    ]
+
+    ## Enables or disables the subscription.
+    enable = true
+
+    ## [optional] The name of the measurement. (default: "ctrlx")
+    # measurement = "ctrlx"
+
+    ## [optional] The interval in which messages are send to the client. (default: 1s)
+    ## Higher values reduce load on network by queuing samples on server side and sending as a single TCP packet.
+    # publish_interval = "1s"
+
+    ## [optional] The interval a "keepalive" message is sent if no change of data occurs. (default: 60s)
+    ## Only used internally to detect broken network connections.
+    # keep_alive_interval = "60s"
+
+    ## [optional] The interval an "error" message is sent if an error was received from a node. (default: 10s)
+    ## Higher values reduce load on output target and network in case of errors by limiting frequency of error messages.
+    # error_interval = "20s"
+
+    ## [optional] The interval that defines the fastest rate at which the node values should be sampled and values captured. (default: 1s)
+    ## The sampling frequency should be adjusted to the dynamics of the signal to be sampled.
+    ## Higher sampling frequence increases load on ctrlX Data Layer.
+    ## The sampling frequency can be higher, than the publish interval. Captured samples are put in a queue and sent in publish interval.
+    ## Note: The minimum sampling interval can be overruled by a global setting in the ctrlX Data Layer configuration ('').
+    # sampling_interval = "1s"
+
```

Comment:
Yes, we are still in the subscription. We will try to make this more clear in the documentation. From our perspective it is necessary to have all this settings, due to the nature of how PLC and automation data is structured.
Also we looked into https://github.com/influxdata/telegraf/blob/master/docs/CONFIGURATION.md#plugins but could not map the data layer settings to those listed there.

Reply:
I think part of the confusion is that it was not clear to me we are talking about a service plugin. Anyway, how likely is it that every `subscription` has a different setting? Would it be an option to define multiple plugin instances?

Reply:
It is not uncommon to have multiple subscriptions, with different settings on each. E.g. one for machine data, the other one for process data and another one for product data. All have different dynamics. Being forced to use multiple plugin-instances would lead to duplicated credentials, thus we would like keep current configuration.

Reply:
Fair enough. Can you maybe add this as a comment? Either to the config option or as an own section in the README?